pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -958,6 +959,47 @@ func TestDeleteReplicaSetOrphan(t *testing.T) {
 	}
 }
 
+func BenchmarkGetPodMapForDeployment(b *testing.B) {
+	f := newFixture(b)
+
+	d := newDeployment("foo", 1, nil, nil, nil, map[string]string{"foo": "bar"})
+
+	rs1 := newReplicaSet(d, "rs1", 1)
+	rs2 := newReplicaSet(d, "rs2", 1)
+
+	var pods []*v1.Pod
+	var objects []runtime.Object
+	for i := 0; i < 100; i++ {
+		p1, p2 := generatePodFromRS(rs1), generatePodFromRS(rs2)
+		p1.Name, p2.Name = p1.Name+fmt.Sprintf("-%d", i), p2.Name+fmt.Sprintf("-%d", i)
+		pods = append(pods, p1, p2)
+		objects = append(objects, p1, p2)
+	}
+
+	f.dLister = append(f.dLister, d)
+	f.rsLister = append(f.rsLister, rs1, rs2)
+	f.podLister = append(f.podLister, pods...)
+	f.objects = append(f.objects, d, rs1, rs2)
+	f.objects = append(f.objects, objects...)
+
+	// Start the fixture.
+	c, informers, err := f.newController()
+	if err != nil {
+		b.Fatalf("error creating Deployment controller: %v", err)
+	}
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	informers.Start(stopCh)
+
+	b.ReportAllocs()
+	for n := 0; n < b.N; n++ {
+		_, err := c.getPodMapForDeployment(d, f.rsLister)
```

Comment:
I'm wondering if go won't optimize it - maybe we should check the result (e.g. check the size of map or sth)?

Reply:
Added, but it doesn't seem to be making any difference.